### PR TITLE
Tests with @Rule still use JUnit 4 vintage

### PR DESCRIPTION
**Pull request by schlosna**
## Before this PR
JUnit 5 excavator https://github.com/palantir/conjure-java-runtime/pull/2485 is stuck.
[2093 tests complete on `develop` check task](https://app.circleci.com/pipelines/github/palantir/conjure-java-runtime/5569/workflows/6947e1f3-f449-4a6b-ab03-7595e75f8d23/jobs/26608/tests)

## After this PR
Merged origin/develop into this branch and updated tests still using `@Rule` to still use JUnit 4 vintage. This is a smaller change than #2495 that completely eradicates JUnit 4, so targeting this sooner to unblock excavator.

[2093 tests complete on this branch's `check` task](https://app.circleci.com/pipelines/github/palantir/conjure-java-runtime/5571/workflows/40a19a74-e065-4e5b-9f2b-20be7d8ed830/jobs/26612/tests)
==COMMIT_MSG==
Tests with @Rule still use JUnit 4 vintage
==COMMIT_MSG==

### CODE REVIEW

File: changelog/7.43.0/pr-2571.v2.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+type: improvement
+improvement:
+  description: Json parser string length metrics are tagged by format
+  links:
+  - https://github.com/palantir/conjure-java-runtime/pull/2571
```
Comment (carterkozak) [inline changelog/7.43.0/pr-2571.v2.yml:5]: Looks like something went sideways in a rebase/merge

Comment (schlosna) [reply changelog/7.43.0/pr-2571.v2.yml:5]: Ugh, the merge of origin/develop seems to have gone sideways, I may just put together a separate PR

Comment (schlosna) [reply changelog/7.43.0/pr-2571.v2.yml:5]: Moving to https://github.com/palantir/conjure-java-runtime/pull/2616

**Comment by schlosna** [comment]
Moving to #2616

**Pull request closed without merging**